packages/hoppscotch-common/src/pages/index.vue

```diff
@@ -61,6 +62,45 @@
         <HttpSidebar />
       </template>
     </AppPaneLayout>
+    <HoppSmartModal
+      v-if="renamingReqName"
+      dialog
+      :title="t('request.name')"
+      @close="renamingReqName = false"
+    >
+      <template #body>
+        <div class="flex flex-col">
+          <input
+            id="reqName"
+            v-model="reqName"
+            v-focus
+            class="input floating-input"
+            placeholder=" "
+            type="text"
+            autocomplete="off"
+            @keyup.enter="renameReqName"
+          />
+          <label for="reqName">
+            {{ t("action.name") }}
+          </label>
+        </div>
+      </template>
+      <template #footer>
+        <span class="flex space-x-2">
+          <HoppButtonPrimary
+            :label="t('action.save')"
+            outline
+            @click="renameReqName"
+          />
+          <HoppButtonSecondary
+            :label="t('action.cancel')"
+            outline
+            filled
+            @click="renamingReqName = false"
+          />
+        </span>
+      </template>
+    </HoppSmartModal>
```

Comment:
can we move this to a modal component or we can reuse the collections/EditRequest.vue component.

Reply:
I too recommend reusing the collections/EditRequest.vue component if feasible.